src/seedu/addressbook/AddressBook.java

```diff
@@ -339,8 +345,9 @@ private static boolean hasValidParentDirectory(Path filePath) {
      * If a file already exists, it must be a regular file.
      */
     private static boolean hasValidFileName(Path filePath) {
-        return filePath.getFileName().toString().lastIndexOf('.') > 0
-                && (!Files.exists(filePath) || Files.isRegularFile(filePath));
+    	boolean isValidFileName = filePath.getFileName().toString().lastIndexOf('.') > 0;
+    	boolean isValidFile = !Files.exists(filePath) || Files.isRegularFile(filePath);
+        return isValidFileName && isValidFile;
```

Comment:
Inconsistency indentation for `return`, make sure your code have consistent indentation. :) Good job for put the complicated functions into boolean variables. BTW, you can also create a function instead of using variables, as you can use them in other functions, if needed.